GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived test failure

https://ci.servicetalk.io/job/servicetalk-java11-prb/1644/testReport/junit/io.servicetalk.http.netty/GracefulConnectionClosureHandlingTest/closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived_4__protocol_HTTP_1_initiateClosureFromClient_false_useUds_false_viaProxy_true_/

```
io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[4: protocol=HTTP_1 initiateClosureFromClient=false useUds=false viaProxy=true]

Error Message
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Stacktrace
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:401)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Standard Output
2020-10-21 22:41:52,542              proxy-tunnel-38-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:41:53,005              proxy-tunnel-39-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:41:53,554              proxy-tunnel-40-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:129) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:41:53,900              proxy-tunnel-41-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:41:54,236              proxy-tunnel-42-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:41:54,734              proxy-tunnel-43-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:41:55,344              proxy-tunnel-44-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:41:55,856              proxy-tunnel-45-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:41:57,508              proxy-tunnel-46-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:41:57,934              proxy-tunnel-47-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:41:58,438              proxy-tunnel-48-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:41:58,783              proxy-tunnel-49-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:129) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:41:59,195                    Test worker [ERROR] ResourceLeakDetector           - LEAK: ByteBuf.release() was not called before it's garbage-collected. See https://netty.io/wiki/reference-counted-objects.html for more information.
Recent access records: 
Created at:
	io.netty.buffer.PooledByteBufAllocator.newDirectBuffer(PooledByteBufAllocator.java:385)
	io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:187)
	io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:178)
	io.servicetalk.http.netty.HttpObjectEncoder.write(HttpObjectEncoder.java:127)
	io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:717)
	io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:709)
	io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:792)
	io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:702)
	io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1015)
	io.netty.channel.AbstractChannel.write(AbstractChannel.java:289)
	io.servicetalk.transport.netty.internal.WriteStreamSubscriber$AllWritesPromise.writeNext(WriteStreamSubscriber.java:340)
	io.servicetalk.transport.netty.internal.WriteStreamSubscriber.doWrite(WriteStreamSubscriber.java:169)
	io.servicetalk.transport.netty.internal.WriteStreamSubscriber.onNext(WriteStreamSubscriber.java:160)
	io.servicetalk.concurrent.api.ContextPreservingSubscriptionSubscriber.onNext(ContextPreservingSubscriptionSubscriber.java:39)
	io.servicetalk.transport.netty.internal.Flush$FlushSubscriber.onNext(Flush.java:116)
	io.servicetalk.concurrent.api.RedoPublisher$RedoSubscriber.onNext(RedoPublisher.java:104)
	io.servicetalk.concurrent.api.SingleFlatMapPublisher$SubscriberImpl.onNext(SingleFlatMapPublisher.java:101)
	io.servicetalk.concurrent.api.ContextPreservingSubscriber.onNext(ContextPreservingSubscriber.java:72)
	io.servicetalk.concurrent.api.ContextPreservingSubscriptionSubscriber.onNext(ContextPreservingSubscriptionSubscriber.java:39)
	io.servicetalk.concurrent.internal.TaskBasedSignalOffloader$OffloadedSubscriptionSubscriber.onNext(TaskBasedSignalOffloader.java:799)
	io.servicetalk.concurrent.api.ContextPreservingSubscriber.onNext(ContextPreservingSubscriber.java:72)
	io.servicetalk.concurrent.api.PublisherConcatWithCompletable$ConcatSubscriber.onNext(PublisherConcatWithCompletable.java:69)
	io.servicetalk.concurrent.internal.TaskBasedSignalOffloader$OffloadedSubscriptionSubscriber.onNext(TaskBasedSignalOffloader.java:799)
	io.servicetalk.concurrent.api.ContextPreservingSubscriber.onNext(ContextPreservingSubscriber.java:72)
	io.servicetalk.concurrent.api.PublisherConcatWithCompletable$ConcatSubscriber.onNext(PublisherConcatWithCompletable.java:69)
	io.servicetalk.concurrent.internal.TaskBasedSignalOffloader$OffloadedSubscriptionSubscriber.onNext(TaskBasedSignalOffloader.java:799)
	io.servicetalk.concurrent.api.ContextPreservingSubscriptionSubscriber.onNext(ContextPreservingSubscriptionSubscriber.java:39)
	io.servicetalk.concurrent.api.SingleFlatMapPublisher$SubscriberImpl.onNext(SingleFlatMapPublisher.java:101)
	io.servicetalk.concurrent.api.ContextPreservingSubscriber.onNext(ContextPreservingSubscriber.java:72)
	io.servicetalk.concurrent.api.SingleFlatMapPublisher$SubscriberImpl.onNext(SingleFlatMapPublisher.java:101)
	io.servicetalk.concurrent.api.ContextPreservingSubscriber.onNext(ContextPreservingSubscriber.java:72)
	io.servicetalk.concurrent.api.ContextPreservingSubscriber.onNext(ContextPreservingSubscriber.java:72)
	io.servicetalk.concurrent.api.PublisherConcatWithSingle$ConcatSubscriber.onNext(PublisherConcatWithSingle.java:80)
	io.servicetalk.concurrent.api.ContextPreservingSubscriber.onNext(ContextPreservingSubscriber.java:72)
	io.servicetalk.concurrent.api.ConcatPublisher$ConcatSubscriber.onNext(ConcatPublisher.java:61)
	io.servicetalk.concurrent.api.ContextPreservingSubscriber.onNext(ContextPreservingSubscriber.java:72)
	io.servicetalk.concurrent.internal.ScalarValueSubscription.request(ScalarValueSubscription.java:61)
	io.servicetalk.concurrent.api.SequentialSubscription.request(SequentialSubscription.java:110)
	io.servicetalk.concurrent.api.ContextPreservingSubscription.request(ContextPreservingSubscription.java:46)
	io.servicetalk.concurrent.api.PublisherConcatWithSingle$ConcatSubscriber.request(PublisherConcatWithSingle.java:157)
	io.servicetalk.concurrent.api.ContextPreservingSubscription.request(ContextPreservingSubscription.java:46)
	io.servicetalk.concurrent.internal.DelayedSubscription.request(DelayedSubscription.java:77)
	io.servicetalk.concurrent.api.DelayedCancellableThenSubscription.request(DelayedCancellableThenSubscription.java:46)
	io.servicetalk.concurrent.api.ContextPreservingSubscription.request(ContextPreservingSubscription.java:46)
	io.servicetalk.concurrent.internal.DelayedSubscription.request(DelayedSubscription.java:77)
	io.servicetalk.concurrent.api.DelayedCancellableThenSubscription.request(DelayedCancellableThenSubscription.java:46)
	io.servicetalk.concurrent.api.ContextPreservingSubscription.request(ContextPreservingSubscription.java:46)
	io.servicetalk.concurrent.internal.TaskBasedSignalOffloader$OffloadedSubscription.run(TaskBasedSignalOffloader.java:220)
	io.servicetalk.concurrent.api.ContextPreservingRunnable.run(ContextPreservingRunnable.java:43)
	java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	java.base/java.lang.Thread.run(Thread.java:834)
2020-10-21 22:41:59,429              proxy-tunnel-50-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:41:59,916              proxy-tunnel-51-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:42:00,402              proxy-tunnel-52-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-21 22:42:01,006              proxy-tunnel-53-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

https://ci.servicetalk.io/job/servicetalk-java11-prb/1650/testReport/junit/io.servicetalk.http.netty/GracefulConnectionClosureHandlingTest/closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived_8__protocol_HTTP_2_initiateClosureFromClient_false_useUds_true_viaProxy_false_/

```
io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[8: protocol=HTTP_2 initiateClosureFromClient=false useUds=true viaProxy=false]

Error Message
java.lang.AssertionError: 
Expected: an instance of java.nio.channels.ClosedChannelException
     but: <io.netty.channel.unix.Errors$NativeIoException: writevAddresses(..) failed: Broken pipe> is a io.netty.channel.unix.Errors$NativeIoException
Stacktrace
java.lang.AssertionError: 
Expected: an instance of java.nio.channels.ClosedChannelException
     but: <io.netty.channel.unix.Errors$NativeIoException: writevAddresses(..) failed: Broken pipe> is a io.netty.channel.unix.Errors$NativeIoException
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.assertClosedChannelException(GracefulConnectionClosureHandlingTest.java:496)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:422)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Standard Output
2020-10-29 01:53:02,440              proxy-tunnel-38-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:129) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:02,846              proxy-tunnel-39-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:03,447              proxy-tunnel-40-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:03,813              proxy-tunnel-41-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:04,203              proxy-tunnel-42-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:04,786              proxy-tunnel-43-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:05,327              proxy-tunnel-44-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:05,739              proxy-tunnel-45-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:07,850              proxy-tunnel-46-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:08,332              proxy-tunnel-47-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:08,981              proxy-tunnel-48-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:09,493              proxy-tunnel-49-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:09,991              proxy-tunnel-50-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:10,740              proxy-tunnel-51-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:11,353              proxy-tunnel-52-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-29 01:53:12,078              proxy-tunnel-53-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

https://ci.servicetalk.io/job/servicetalk-java11-prb/1652/testReport/junit/io.servicetalk.http.netty/GracefulConnectionClosureHandlingTest/closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived_4__protocol_HTTP_1_initiateClosureFromClient_false_useUds_false_viaProxy_true_/

```
io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[4: protocol=HTTP_1 initiateClosureFromClient=false useUds=false viaProxy=true]

Error Message
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Stacktrace
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:401)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Standard Output
2020-10-31 02:02:58,598              proxy-tunnel-38-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:129) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:02:59,053              proxy-tunnel-39-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:02:59,664              proxy-tunnel-40-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:129) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:03:00,071              proxy-tunnel-41-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:03:00,431              proxy-tunnel-42-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:03:00,966              proxy-tunnel-43-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:03:01,569              proxy-tunnel-44-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:03:02,014              proxy-tunnel-45-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:03:04,402              proxy-tunnel-46-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:03:04,895              proxy-tunnel-47-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:03:05,497              proxy-tunnel-48-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:03:06,014              proxy-tunnel-49-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:129) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:03:06,505              proxy-tunnel-50-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:03:07,176              proxy-tunnel-51-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:03:07,794              proxy-tunnel-52-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-10-31 02:03:08,471              proxy-tunnel-53-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

https://ci.servicetalk.io/job/servicetalk-java8-prb/1665/testReport/junit/io.servicetalk.http.netty/GracefulConnectionClosureHandlingTest/closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived_4__protocol_HTTP_1_initiateClosureFromClient_false_useUds_false_viaProxy_true_/

```
io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[4: protocol=HTTP_1 initiateClosureFromClient=false useUds=false viaProxy=true]

Error Message
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Stacktrace
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:401)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Standard Output
2020-11-04 20:04:44,909             proxy-tunnel-564-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:118) ~[?:1.8.0_232]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:134) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:45,221             proxy-tunnel-565-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:45,604             proxy-tunnel-566-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:118) ~[?:1.8.0_232]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:134) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:45,978             proxy-tunnel-567-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:46,157             proxy-tunnel-568-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:46,601             proxy-tunnel-569-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:46,943             proxy-tunnel-570-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:47,349             proxy-tunnel-571-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:49,026             proxy-tunnel-572-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:49,317             proxy-tunnel-573-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:49,734             proxy-tunnel-574-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:50,008             proxy-tunnel-575-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:118) ~[?:1.8.0_232]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:134) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:50,322             proxy-tunnel-576-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:50,711             proxy-tunnel-577-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:51,018             proxy-tunnel-578-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-04 20:04:51,353             proxy-tunnel-579-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
```

https://ci.servicetalk.io/job/servicetalk-java11-prb/1680/testReport/junit/io.servicetalk.http.netty/GracefulConnectionClosureHandlingTest/closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived_4__protocol_HTTP_1_initiateClosureFromClient_false_useUds_false_viaProxy_true_/

```
io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[4: protocol=HTTP_1 initiateClosureFromClient=false useUds=false viaProxy=true]

Error Message
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Stacktrace
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:402)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Standard Output
2020-11-11 04:21:18,269             proxy-tunnel-887-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:129) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:18,591             proxy-tunnel-888-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:18,929             proxy-tunnel-889-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:129) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:19,214             proxy-tunnel-890-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:19,371             proxy-tunnel-891-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:19,668             proxy-tunnel-892-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:19,953             proxy-tunnel-893-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:20,222             proxy-tunnel-894-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:21,347             proxy-tunnel-895-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:21,762             proxy-tunnel-896-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:22,222             proxy-tunnel-897-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:22,627             proxy-tunnel-898-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:129) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:23,051             proxy-tunnel-899-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:23,465             proxy-tunnel-900-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:23,843             proxy-tunnel-901-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-11 04:21:24,271             proxy-tunnel-902-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]

```

https://ci.servicetalk.io/job/servicetalk-java8-prb/1707/testReport/junit/io.servicetalk.http.netty/GracefulConnectionClosureHandlingTest/closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived_4__protocol_HTTP_1_initiateClosureFromClient_false_useUds_false_viaProxy_true_/

```
io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[4: protocol=HTTP_1 initiateClosureFromClient=false useUds=false viaProxy=true]

Error Message
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Stacktrace
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:402)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Standard Output
2020-11-18 21:56:27,418             proxy-tunnel-451-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:27,679             proxy-tunnel-452-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:28,062             proxy-tunnel-453-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:118) ~[?:1.8.0_232]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:134) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:28,370             proxy-tunnel-454-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:28,624             proxy-tunnel-455-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:28,931             proxy-tunnel-456-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:29,206             proxy-tunnel-457-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:29,531             proxy-tunnel-458-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:31,138             proxy-tunnel-459-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:31,466             proxy-tunnel-460-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:31,806             proxy-tunnel-461-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:32,064             proxy-tunnel-462-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:118) ~[?:1.8.0_232]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:134) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:32,377             proxy-tunnel-463-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:32,723             proxy-tunnel-464-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:32,971             proxy-tunnel-465-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 21:56:33,214             proxy-tunnel-466-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
```

https://ci.servicetalk.io/job/servicetalk-java8-prb/1711/testReport/junit/io.servicetalk.http.netty/GracefulConnectionClosureHandlingTest/closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived_9__protocol_HTTP_2_initiateClosureFromClient_false_useUds_false_viaProxy_false_/

```
io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[9: protocol=HTTP_2 initiateClosureFromClient=false useUds=false viaProxy=false]

Error Message
java.util.concurrent.ExecutionException: io.servicetalk.transport.netty.internal.StacklessClosedChannelException
Stacktrace
java.util.concurrent.ExecutionException: io.servicetalk.transport.netty.internal.StacklessClosedChannelException
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:402)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.servicetalk.transport.netty.internal.StacklessClosedChannelException
	at io.servicetalk.transport.netty.internal.NettyChannelPublisher.channelInboundClosed(Unknown Source)
Standard Output
2020-11-18 22:47:18,504             proxy-tunnel-450-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:118) ~[?:1.8.0_232]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:134) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:18,866             proxy-tunnel-451-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:19,238             proxy-tunnel-452-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:118) ~[?:1.8.0_232]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:134) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:19,566             proxy-tunnel-453-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:19,815             proxy-tunnel-454-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:20,161             proxy-tunnel-455-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:20,487             proxy-tunnel-456-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:20,869             proxy-tunnel-457-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:22,594             proxy-tunnel-458-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:22,841             proxy-tunnel-459-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:23,178             proxy-tunnel-460-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:23,441             proxy-tunnel-461-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:23,671             proxy-tunnel-462-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:24,002             proxy-tunnel-463-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:24,256             proxy-tunnel-464-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:24,513             proxy-tunnel-465-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:27,539                 server-io-24-8 [WARN ] Http2FrameCodec                - Stream exception thrown for unknown stream 5.
io.netty.handler.codec.http2.Http2Exception$StreamException: Cannot create stream 5 greater than Last-Stream-ID 3 from GOAWAY.
	at io.netty.handler.codec.http2.Http2Exception.streamError(Http2Exception.java:152) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2Connection$DefaultEndpoint.checkNewStreamAllowed(DefaultHttp2Connection.java:874) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2Connection$DefaultEndpoint.createStream(DefaultHttp2Connection.java:748) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2Connection$DefaultEndpoint.createStream(DefaultHttp2Connection.java:668) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder$FrameReadListener.onHeadersRead(DefaultHttp2ConnectionDecoder.java:330) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder$FrameReadListener.onHeadersRead(DefaultHttp2ConnectionDecoder.java:321) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.Http2InboundFrameLogger$1.onHeadersRead(Http2InboundFrameLogger.java:56) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2FrameReader$2.processFragment(DefaultHttp2FrameReader.java:483) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2FrameReader.readHeadersFrame(DefaultHttp2FrameReader.java:491) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2FrameReader.processPayloadState(DefaultHttp2FrameReader.java:254) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2FrameReader.readFrame(DefaultHttp2FrameReader.java:160) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.Http2InboundFrameLogger.readFrame(Http2InboundFrameLogger.java:41) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder.decodeFrame(DefaultHttp2ConnectionDecoder.java:174) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DecoratingHttp2ConnectionDecoder.decodeFrame(DecoratingHttp2ConnectionDecoder.java:63) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.Http2ConnectionHandler$FrameDecoder.decode(Http2ConnectionHandler.java:378) [netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.Http2ConnectionHandler.decode(Http2ConnectionHandler.java:438) [netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:501) [netty-codec-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:440) [netty-codec-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276) [netty-codec-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.servicetalk.transport.netty.internal.ServiceTalkWireLogger.channelRead(ServiceTalkWireLogger.java:151) [servicetalk-transport-netty-internal-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:795) [netty-transport-native-epoll-4.1.54.Final-linux-x86_64.jar:4.1.54.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:480) [netty-transport-native-epoll-4.1.54.Final-linux-x86_64.jar:4.1.54.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) [netty-transport-native-epoll-4.1.54.Final-linux-x86_64.jar:4.1.54.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) [netty-common-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.54.Final.jar:4.1.54.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-18 22:47:27,544                 server-io-24-8 [INFO ] DefaultHttp2ConnectionDecoder  - [id: 0x7614b37b, L:/127.0.0.1:40872 - R:/127.0.0.1:49274] ignoring DATA frame for stream 5. Stream sent after GOAWAY sent
2020-11-18 22:47:27,550          server-executor-448-5 [ERROR] NettyHttpServer$NettyHttpServerConnection - Internal server error service=io.servicetalk.http.api.AsyncContextAwareHttpServiceFilter$1@79052d88 connection=NettyHttpServerConnection([id: 0x7614b37b, L:/127.0.0.1:40872 ! R:/127.0.0.1:49274](H2 - 3))
java.lang.InterruptedException: null
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:998) ~[?:1.8.0_232]
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304) ~[?:1.8.0_232]
	at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:231) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.lambda$new$3(GracefulConnectionClosureHandlingTest.java:181) ~[test/:?]
	at io.servicetalk.http.api.BlockingStreamingToStreamingService$1.handleSubscribe(BlockingStreamingToStreamingService.java:140) ~[servicetalk-http-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.Single.handleSubscribe(Single.java:1954) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.Single.delegateSubscribe(Single.java:1910) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.ResumeSingle.handleSubscribe(ResumeSingle.java:51) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.Single.delegateSubscribe(Single.java:1910) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.SingleFlatMapPublisher.handleSubscribe(SingleFlatMapPublisher.java:44) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.PublisherSubscribeShareContext.handleSubscribe(PublisherSubscribeShareContext.java:34) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.Publisher.lambda$subscribeWithContext$10(Publisher.java:2833) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.ContextPreservingConsumer.accept(ContextPreservingConsumer.java:40) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.NoopOffloader.offloadSubscribe(NoopOffloader.java:68) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.Publisher.subscribeWithContext(Publisher.java:2832) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.Publisher.subscribeInternal(Publisher.java:2577) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.SingleFlatMapPublisher$SubscriberImpl.onSuccess(SingleFlatMapPublisher.java:95) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.ResumeSingle$ResumeSubscriber.onSuccess(ResumeSingle.java:90) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.ContextPreservingSingleSubscriber.onSuccess(ContextPreservingSingleSubscriber.java:75) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.AbstractSubmitSingle.lambda$handleSubscribe$0(AbstractSubmitSingle.java:49) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.ContextPreservingRunnable.run(ContextPreservingRunnable.java:43) [servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
```

https://ci.servicetalk.io/job/servicetalk-java8-prb/1740/testReport/junit/io.servicetalk.http.netty/GracefulConnectionClosureHandlingTest/closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived_8__protocol_HTTP_2_initiateClosureFromClient_false_useUds_true_viaProxy_false_/

```
io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[8: protocol=HTTP_2 initiateClosureFromClient=false useUds=true viaProxy=false]

Error Message
java.lang.AssertionError: 
Expected: an instance of java.nio.channels.ClosedChannelException
     but: <io.netty.handler.codec.http2.Http2Exception$StreamException: Cannot create stream 5 greater than Last-Stream-ID 3 from GOAWAY.> is a io.netty.handler.codec.http2.Http2Exception$StreamException
Stacktrace
java.lang.AssertionError: 
Expected: an instance of java.nio.channels.ClosedChannelException
     but: <io.netty.handler.codec.http2.Http2Exception$StreamException: Cannot create stream 5 greater than Last-Stream-ID 3 from GOAWAY.> is a io.netty.handler.codec.http2.Http2Exception$StreamException
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.assertClosedChannelException(GracefulConnectionClosureHandlingTest.java:497)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:423)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Standard Output
2020-11-21 20:03:40,287             proxy-tunnel-304-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:118) ~[?:1.8.0_232]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:134) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:40,493             proxy-tunnel-305-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:40,753             proxy-tunnel-306-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:40,945             proxy-tunnel-307-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:41,035             proxy-tunnel-308-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:41,197             proxy-tunnel-309-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:41,358             proxy-tunnel-310-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:41,518             proxy-tunnel-311-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:42,211             proxy-tunnel-312-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:42,438             proxy-tunnel-313-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:42,666             proxy-tunnel-314-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:42,826             proxy-tunnel-315-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:42,947             proxy-tunnel-316-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:43,198             proxy-tunnel-317-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:43,426             proxy-tunnel-318-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:43,670             proxy-tunnel-319-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 20:03:44,620                 client-io-23-4 [WARN ] Http2FrameCodec                - Stream exception thrown for unknown stream 5.
io.netty.handler.codec.http2.Http2Exception$StreamException: Cannot create stream 5 greater than Last-Stream-ID 3 from GOAWAY.
	at io.netty.handler.codec.http2.Http2Exception.streamError(Http2Exception.java:152) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2Connection$DefaultEndpoint.checkNewStreamAllowed(DefaultHttp2Connection.java:874) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2Connection$DefaultEndpoint.createStream(DefaultHttp2Connection.java:748) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2Connection$DefaultEndpoint.createStream(DefaultHttp2Connection.java:668) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2ConnectionEncoder.writeHeaders0(DefaultHttp2ConnectionEncoder.java:200) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2ConnectionEncoder.writeHeaders(DefaultHttp2ConnectionEncoder.java:150) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DecoratingHttp2FrameWriter.writeHeaders(DecoratingHttp2FrameWriter.java:45) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.Http2FrameCodec.writeHeadersFrame(Http2FrameCodec.java:418) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.Http2FrameCodec.write(Http2FrameCodec.java:301) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:717) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:709) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:792) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:702) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.AbstractHttp2StreamChannel.write0(AbstractHttp2StreamChannel.java:1099) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.AbstractHttp2StreamChannel$Http2ChannelUnsafe.writeHttp2StreamFrame(AbstractHttp2StreamChannel.java:952) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.AbstractHttp2StreamChannel$Http2ChannelUnsafe.write(AbstractHttp2StreamChannel.java:923) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.write(DefaultChannelPipeline.java:1367) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:717) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:709) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:792) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:702) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.servicetalk.http.netty.H2ToStH1ClientDuplexHandler.write(H2ToStH1ClientDuplexHandler.java:89) ~[main/:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:717) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:709) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:792) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:702) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1015) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.AbstractHttp2StreamChannel.write(AbstractHttp2StreamChannel.java:480) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.servicetalk.transport.netty.internal.WriteStreamSubscriber$AllWritesPromise.writeNext(WriteStreamSubscriber.java:340) ~[servicetalk-transport-netty-internal-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.transport.netty.internal.WriteStreamSubscriber.doWrite(WriteStreamSubscriber.java:169) ~[servicetalk-transport-netty-internal-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.transport.netty.internal.WriteStreamSubscriber.lambda$onNext$2(WriteStreamSubscriber.java:156) ~[servicetalk-transport-netty-internal-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) [netty-common-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472) [netty-common-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384) [netty-transport-native-epoll-4.1.54.Final-linux-x86_64.jar:4.1.54.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) [netty-common-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.54.Final.jar:4.1.54.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
```

https://ci.servicetalk.io/job/servicetalk-java11-prb/1741/testReport/junit/io.servicetalk.http.netty/GracefulConnectionClosureHandlingTest/closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived_4__protocol_HTTP_1_initiateClosureFromClient_false_useUds_false_viaProxy_true_/

```
io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[4: protocol=HTTP_1 initiateClosureFromClient=false useUds=false viaProxy=true]

Error Message
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Stacktrace
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:402)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Standard Output
2020-11-21 20:22:15,072             proxy-tunnel-680-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:129) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:15,448             proxy-tunnel-681-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:15,785             proxy-tunnel-682-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:16,104             proxy-tunnel-683-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:16,499             proxy-tunnel-684-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:16,825             proxy-tunnel-685-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:17,250             proxy-tunnel-686-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:17,596             proxy-tunnel-687-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:18,481             proxy-tunnel-688-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:18,772             proxy-tunnel-689-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:19,041             proxy-tunnel-690-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:19,335             proxy-tunnel-691-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:129) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:19,600             proxy-tunnel-692-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:20,001             proxy-tunnel-693-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:20,365             proxy-tunnel-694-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-21 20:22:20,762             proxy-tunnel-695-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

https://ci.servicetalk.io/job/servicetalk-java8-prb/1743/testReport/junit/io.servicetalk.http.netty/GracefulConnectionClosureHandlingTest/closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived_4__protocol_HTTP_1_initiateClosureFromClient_false_useUds_false_viaProxy_true_/

```
io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[4: protocol=HTTP_1 initiateClosureFromClient=false useUds=false viaProxy=true]

Error Message
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Stacktrace
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:402)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Standard Output
2020-11-21 21:53:20,455             proxy-tunnel-304-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:118) ~[?:1.8.0_232]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:134) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:20,791             proxy-tunnel-305-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:21,160             proxy-tunnel-306-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:21,498             proxy-tunnel-307-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:21,772             proxy-tunnel-308-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:22,095             proxy-tunnel-309-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:22,511             proxy-tunnel-310-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:22,916             proxy-tunnel-311-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:24,760             proxy-tunnel-312-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:25,053             proxy-tunnel-313-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:25,403             proxy-tunnel-314-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:25,721             proxy-tunnel-315-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:118) ~[?:1.8.0_232]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:134) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:25,898             proxy-tunnel-316-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:26,247             proxy-tunnel-317-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:26,534             proxy-tunnel-318-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-11-21 21:53:26,807             proxy-tunnel-319-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
```

https://ci.servicetalk.io/job/servicetalk-java11-prb/1750/testReport/junit/io.servicetalk.http.netty/GracefulConnectionClosureHandlingTest/closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived_4__protocol_HTTP_1_initiateClosureFromClient_false_useUds_false_viaProxy_true_/

```
io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[4: protocol=HTTP_1 initiateClosureFromClient=false useUds=false viaProxy=true]

Error Message
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Stacktrace
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:402)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
Standard Output
2020-11-24 00:59:59,992             proxy-tunnel-680-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:00,328             proxy-tunnel-681-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:00,740             proxy-tunnel-682-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:129) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:01,325             proxy-tunnel-683-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:01,689             proxy-tunnel-684-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:02,133             proxy-tunnel-685-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:02,651             proxy-tunnel-686-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:03,240             proxy-tunnel-687-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:04,703             proxy-tunnel-688-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:05,142             proxy-tunnel-689-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:05,516             proxy-tunnel-690-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:05,945             proxy-tunnel-691-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113) ~[?:?]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:129) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:06,391             proxy-tunnel-692-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:06,806             proxy-tunnel-693-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:07,331             proxy-tunnel-694-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2020-11-24 01:00:07,809             proxy-tunnel-695-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:?]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:115) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:168) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:140) ~[?:?]
	at java.net.SocketInputStream.read(SocketInputStream.java:200) ~[?:?]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$startProxy$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

https://ci.servicetalk.io/job/servicetalk-java8-prb/1780/testReport/junit/io.servicetalk.http.netty/GracefulConnectionClosureHandlingTest/closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived_8__protocol_HTTP_2_initiateClosureFromClient_false_useUds_true_viaProxy_false_/

```
io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[8: protocol=HTTP_2 initiateClosureFromClient=false useUds=true viaProxy=false]

Error Message
java.util.concurrent.ExecutionException: io.servicetalk.transport.netty.internal.StacklessClosedChannelException
Stacktrace
java.util.concurrent.ExecutionException: io.servicetalk.transport.netty.internal.StacklessClosedChannelException
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:94)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:402)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.servicetalk.transport.netty.internal.StacklessClosedChannelException
	at io.servicetalk.transport.netty.internal.NettyChannelPublisher.channelInboundClosed(Unknown Source)
Standard Output
2020-12-07 20:04:48,687             proxy-tunnel-448-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:118) ~[?:1.8.0_232]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:134) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:48,979             proxy-tunnel-449-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:49,254             proxy-tunnel-450-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:118) ~[?:1.8.0_232]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:134) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:145) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:49,519             proxy-tunnel-451-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:49,730             proxy-tunnel-452-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:50,007             proxy-tunnel-453-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:50,257             proxy-tunnel-454-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:50,555             proxy-tunnel-455-3 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$handleRequest$3(ProxyTunnel.java:134) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:52,221             proxy-tunnel-456-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:52,533             proxy-tunnel-457-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:52,843             proxy-tunnel-458-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:53,095             proxy-tunnel-459-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:53,303             proxy-tunnel-460-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:53,618             proxy-tunnel-461-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:53,882             proxy-tunnel-462-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:54,245             proxy-tunnel-463-2 [ERROR] ProxyTunnel                    - Proxy exception
java.net.SocketException: Socket closed
	at java.net.SocketInputStream.socketRead0(Native Method) ~[?:1.8.0_232]
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:171) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_232]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.ProxyTunnel.copyStream(ProxyTunnel.java:144) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.handleRequest(ProxyTunnel.java:135) ~[test/:?]
	at io.servicetalk.http.netty.ProxyTunnel.lambda$null$0(ProxyTunnel.java:75) ~[test/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:56,659                 server-io-21-8 [WARN ] Http2FrameCodec                - Stream exception thrown for unknown stream 5.
io.netty.handler.codec.http2.Http2Exception$StreamException: Cannot create stream 5 greater than Last-Stream-ID 3 from GOAWAY.
	at io.netty.handler.codec.http2.Http2Exception.streamError(Http2Exception.java:152) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2Connection$DefaultEndpoint.checkNewStreamAllowed(DefaultHttp2Connection.java:874) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2Connection$DefaultEndpoint.createStream(DefaultHttp2Connection.java:748) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2Connection$DefaultEndpoint.createStream(DefaultHttp2Connection.java:668) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder$FrameReadListener.onHeadersRead(DefaultHttp2ConnectionDecoder.java:330) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder$FrameReadListener.onHeadersRead(DefaultHttp2ConnectionDecoder.java:321) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.Http2InboundFrameLogger$1.onHeadersRead(Http2InboundFrameLogger.java:56) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2FrameReader$2.processFragment(DefaultHttp2FrameReader.java:483) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2FrameReader.readHeadersFrame(DefaultHttp2FrameReader.java:491) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2FrameReader.processPayloadState(DefaultHttp2FrameReader.java:254) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2FrameReader.readFrame(DefaultHttp2FrameReader.java:160) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.Http2InboundFrameLogger.readFrame(Http2InboundFrameLogger.java:41) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder.decodeFrame(DefaultHttp2ConnectionDecoder.java:174) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.DecoratingHttp2ConnectionDecoder.decodeFrame(DecoratingHttp2ConnectionDecoder.java:63) ~[netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.Http2ConnectionHandler$FrameDecoder.decode(Http2ConnectionHandler.java:378) [netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.http2.Http2ConnectionHandler.decode(Http2ConnectionHandler.java:438) [netty-codec-http2-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:501) [netty-codec-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:440) [netty-codec-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276) [netty-codec-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.servicetalk.transport.netty.internal.ServiceTalkWireLogger.channelRead(ServiceTalkWireLogger.java:151) [servicetalk-transport-netty-internal-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [netty-transport-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:795) [netty-transport-native-epoll-4.1.54.Final-linux-x86_64.jar:4.1.54.Final]
	at io.netty.channel.epoll.EpollDomainSocketChannel$EpollDomainUnsafe.epollInReady(EpollDomainSocketChannel.java:138) [netty-transport-native-epoll-4.1.54.Final-linux-x86_64.jar:4.1.54.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:480) [netty-transport-native-epoll-4.1.54.Final-linux-x86_64.jar:4.1.54.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) [netty-transport-native-epoll-4.1.54.Final-linux-x86_64.jar:4.1.54.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) [netty-common-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.54.Final.jar:4.1.54.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.54.Final.jar:4.1.54.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
2020-12-07 20:04:56,672                 server-io-21-8 [INFO ] DefaultHttp2ConnectionDecoder  - [id: 0x63a63e36] ignoring DATA frame for stream 5. Stream sent after GOAWAY sent
2020-12-07 20:04:56,676          server-executor-446-5 [ERROR] NettyHttpServer$NettyHttpServerConnection - Internal server error service=io.servicetalk.http.api.AsyncContextAwareHttpServiceFilter$1@6d6d9b60 connection=NettyHttpServerConnection([id: 0x63a63e36](H2 - 3))
java.lang.InterruptedException: null
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:998) ~[?:1.8.0_232]
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304) ~[?:1.8.0_232]
	at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:231) ~[?:1.8.0_232]
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.lambda$new$3(GracefulConnectionClosureHandlingTest.java:181) ~[test/:?]
	at io.servicetalk.http.api.BlockingStreamingToStreamingService$1.handleSubscribe(BlockingStreamingToStreamingService.java:140) ~[servicetalk-http-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.Single.handleSubscribe(Single.java:1954) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.Single.delegateSubscribe(Single.java:1910) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.ResumeSingle.handleSubscribe(ResumeSingle.java:51) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.Single.delegateSubscribe(Single.java:1910) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.SingleFlatMapPublisher.handleSubscribe(SingleFlatMapPublisher.java:44) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.PublisherSubscribeShareContext.handleSubscribe(PublisherSubscribeShareContext.java:34) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.Publisher.lambda$subscribeWithContext$11(Publisher.java:2992) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.ContextPreservingConsumer.accept(ContextPreservingConsumer.java:40) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.NoopOffloader.offloadSubscribe(NoopOffloader.java:68) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.Publisher.subscribeWithContext(Publisher.java:2991) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.Publisher.subscribeInternal(Publisher.java:2736) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.SingleFlatMapPublisher$SubscriberImpl.onSuccess(SingleFlatMapPublisher.java:95) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.ResumeSingle$ResumeSubscriber.onSuccess(ResumeSingle.java:90) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.ContextPreservingSingleSubscriber.onSuccess(ContextPreservingSingleSubscriber.java:75) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.AbstractSubmitSingle.lambda$handleSubscribe$0(AbstractSubmitSingle.java:49) ~[servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at io.servicetalk.concurrent.api.ContextPreservingRunnable.run(ContextPreservingRunnable.java:43) [servicetalk-concurrent-api-0.35.0-SNAPSHOT.jar:0.35.0-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_232]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_232]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_232]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
```

https://github.com/apple/servicetalk/pull/1272/checks?check_run_id=1534139256

```
java.util.concurrent.ExecutionException: io.servicetalk.transport.netty.internal.StacklessClosedChannelException
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:402)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.servicetalk.transport.netty.internal.StacklessClosedChannelException
	at io.servicetalk.transport.netty.internal.NettyChannelPublisher.channelInboundClosed(Unknown Source)
```

https://github.com/apple/servicetalk/runs/1704174279

GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[8: protocol=HTTP_2 initiateClosureFromClient=false useUds=true viaProxy=false]

```
java.util.concurrent.ExecutionException: io.servicetalk.transport.netty.internal.StacklessClosedChannelException
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:402)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.servicetalk.transport.netty.internal.StacklessClosedChannelException
	at io.servicetalk.transport.netty.internal.NettyChannelPublisher.channelInboundClosed(Unknown Source)
```

https://github.com/apple/servicetalk/runs/1706118253

```
GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[8: protocol=HTTP_2 initiateClosureFromClient=false useUds=true viaProxy=false]

java.util.concurrent.ExecutionException: io.servicetalk.transport.netty.internal.StacklessClosedChannelException
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:402)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.servicetalk.transport.netty.internal.StacklessClosedChannelException
	at io.servicetalk.transport.netty.internal.NettyChannelPublisher.channelInboundClosed(Unknown Source)
```

https://github.com/apple/servicetalk/runs/1706196323

```
GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[9: protocol=HTTP_2 initiateClosureFromClient=false useUds=false viaProxy=false]

java.lang.AssertionError: 
Expected: an instance of java.nio.channels.ClosedChannelException
     but: <io.servicetalk.http.netty.Http2Exception$RetryableStacklessHttp2Exception: io.netty.handler.codec.http2.Http2Exception$StreamException: Cannot create stream 5 greater than Last-Stream-ID 3 from GOAWAY.> is a io.servicetalk.http.netty.Http2Exception$RetryableStacklessHttp2Exception
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.assertClosedChannelException(GracefulConnectionClosureHandlingTest.java:497)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:423)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:171)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:163)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

https://github.com/apple/servicetalk/runs/1739144715

```
GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[6: protocol=HTTP_2 initiateClosureFromClient=true useUds=true viaProxy=false]

java.util.concurrent.ExecutionException: io.servicetalk.transport.netty.internal.StacklessClosedChannelException
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:402)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:170)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:162)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.servicetalk.transport.netty.internal.StacklessClosedChannelException
	at io.servicetalk.transport.netty.internal.NettyChannelPublisher.channelInboundClosed(Unknown Source)
```

https://github.com/apple/servicetalk/pull/1332/checks?check_run_id=1793614408

```
GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[4: protocol=HTTP_1 initiateClosureFromClient=false useUds=false viaProxy=true]

java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:402)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:170)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:162)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
```

https://github.com/apple/servicetalk/runs/1890581284

```
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:402)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:170)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:162)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
```

https://github.com/apple/servicetalk/pull/1389/checks?check_run_id=1989314572

```
GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[4: protocol=HTTP_1 initiateClosureFromClient=false useUds=false viaProxy=true]

java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:402)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:170)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:162)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
```

https://github.com/apple/servicetalk/runs/2100937525
```
GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[4: protocol=HTTP_1 initiateClosureFromClient=false useUds=false viaProxy=true]

java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:405)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:170)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:162)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
```

https://github.com/apple/servicetalk/runs/2126189094

```
GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[4: protocol=HTTP_1 initiateClosureFromClient=false useUds=false viaProxy=true]

java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.assertResponsePayloadBody(GracefulConnectionClosureHandlingTest.java:467)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:407)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:170)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:162)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
```

https://github.com/apple/servicetalk/runs/2356292385

```
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.assertResponsePayloadBody(GracefulConnectionClosureHandlingTest.java:467)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:407)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:161)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:153)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
```

https://github.com/apple/servicetalk/runs/2395042378

`
 GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived[4: protocol=HTTP_1 initiateClosureFromClient=false useUds=false viaProxy=true] 
`

```
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
	at io.servicetalk.concurrent.api.SourceToFuture.reportGet(SourceToFuture.java:121)
	at io.servicetalk.concurrent.api.SourceToFuture.get(SourceToFuture.java:92)
	at io.servicetalk.http.netty.GracefulConnectionClosureHandlingTest.closePipelinedAfterTwoRequestsSentBeforeAnyResponseReceived(GracefulConnectionClosureHandlingTest.java:408)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:161)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:153)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.netty.channel.unix.Errors$NativeIoException: readAddress(..) failed: Connection reset by peer
```

--- a/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/CloseHandler.java
+++ b/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/CloseHandler.java
@@ -22,6 +22,7 @@ import io.netty.channel.ChannelPromise;
 import io.netty.channel.EventLoop;
 import io.netty.channel.socket.DuplexChannel;
 import io.netty.channel.socket.SocketChannel;
+import io.netty.handler.ssl.SslCloseCompletionEvent;
 
 import java.nio.channels.ClosedChannelException;
 import java.util.function.Consumer;
@@ -128,6 +129,16 @@ public abstract class CloseHandler {
      * @param ctx {@link ChannelHandlerContext}
      */
     abstract void channelClosedOutbound(ChannelHandlerContext ctx);
+
+    /**
+     * Signal {@link Channel} observed {@link SslCloseCompletionEvent#SUCCESS}.
+     * <p>
+     * Received <a href="https://tools.ietf.org/html/rfc5246#section-7.2.1">close_notify</a> alert from the peer.
+     * This message notifies that the sender will not send any more messages on this connection.
+     *
+     * @param ctx {@link ChannelHandlerContext}
+     */
+    abstract void channelCloseNotify(ChannelHandlerContext ctx);
 
     /**
      * Request {@link Channel} inbound close, to be emitted from the {@link EventLoop} for the channel.
@@ -248,6 +259,10 @@ public abstract class CloseHandler {
 
         @Override
         void channelClosedOutbound(final ChannelHandlerContext ctx) {
+        }
+
+        @Override
+        void channelCloseNotify(final ChannelHandlerContext ctx) {
         }
 
         @Override

--- a/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java
+++ b/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java
@@ -643,14 +643,7 @@ public final class DefaultNettyConnection<Read, Write> extends NettyChannelListe
                 connection.channelOutboundListener.channelClosed(StacklessClosedChannelException.newInstance(
                         DefaultNettyConnection.class, "userEventTriggered(ChannelOutputShutdownEvent)"));
             } else if (evt == SslCloseCompletionEvent.SUCCESS) {
-                // Received "close_notify" alert from the peer: https://tools.ietf.org/html/rfc5246#section-7.2.1.
-                // This message notifies that the sender will not send any more messages on this connection.
-
-                // Notify close handler first to enhance error reporting and prevent LB from selecting this connection
-                connection.closeHandler.channelClosedInbound(ctx);
-                // We MUST respond with a "close_notify" alert and close down the connection immediately,
-                // discarding any pending writes.
-                connection.closeHandler.closeChannelOutbound(ctx.channel());
+                connection.closeHandler.channelCloseNotify(ctx);
             } else if (evt == ChannelInputShutdownReadComplete.INSTANCE) {
                 // Notify close handler first to enhance error reporting and prevent LB from selecting this connection
                 connection.closeHandler.channelClosedInbound(ctx);

--- a/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NonPipelinedCloseHandler.java
+++ b/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NonPipelinedCloseHandler.java
@@ -18,6 +18,8 @@ package io.servicetalk.transport.netty.internal;
 import io.netty.channel.Channel;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.channel.ChannelPromise;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.function.Consumer;
 import javax.annotation.Nullable;
@@ -31,6 +33,8 @@ import static io.servicetalk.transport.netty.internal.RequestResponseCloseHandle
 import static java.util.Objects.requireNonNull;
 
 final class NonPipelinedCloseHandler extends CloseHandler {
+    private static final Logger LOGGER = LoggerFactory.getLogger(NonPipelinedCloseHandler.class);
+
     private static final byte READ = 0x1;
     private static final byte WRITE = 0x2;
     private static final byte IN_CLOSED = 0x4;
@@ -62,7 +66,7 @@ final class NonPipelinedCloseHandler extends CloseHandler {
     @Override
     public void protocolPayloadEndInbound(final ChannelHandlerContext ctx) {
         state = unset(state, READ);
-        inboundEventCheckClose(ctx.channel());
+        inboundEventCheckClose(ctx.channel(), closeEvent);
     }
 
     @Override
@@ -75,22 +79,24 @@ final class NonPipelinedCloseHandler extends CloseHandler {
         ctx.pipeline().fireUserEventTriggered(OutboundDataEndEvent.INSTANCE);
         promise.addListener(f -> {
             state = unset(state, WRITE);
-            outboundEventCheckClose(ctx.channel());
+            outboundEventCheckClose(ctx.channel(), closeEvent);
         });
     }
 
     @Override
     public void protocolClosingInbound(final ChannelHandlerContext ctx) {
         state = set(state, IN_CLOSED);
-        storeCloseRequestAndEmit(PROTOCOL_CLOSING_INBOUND);
-        inboundEventCheckClose(ctx.channel());
+        final CloseEvent evt = PROTOCOL_CLOSING_INBOUND;
+        storeCloseRequestAndEmit(evt);
+        inboundEventCheckClose(ctx.channel(), evt);
     }
 
     @Override
     public void protocolClosingOutbound(final ChannelHandlerContext ctx) {
         state = set(state, OUT_CLOSED);
-        storeCloseRequestAndEmit(PROTOCOL_CLOSING_OUTBOUND);
-        outboundEventCheckClose(ctx.channel());
+        final CloseEvent evt = PROTOCOL_CLOSING_OUTBOUND;
+        storeCloseRequestAndEmit(evt);
+        outboundEventCheckClose(ctx.channel(), evt);
     }
 
     @Override
@@ -100,44 +106,57 @@ final class NonPipelinedCloseHandler extends CloseHandler {
 
     @Override
     void channelClosedInbound(final ChannelHandlerContext ctx) {
-        state = unset(set(state, IN_CLOSED), READ);
-        storeCloseRequestAndEmit(CHANNEL_CLOSED_INBOUND);
-        inboundEventCheckClose(ctx.channel());
+        if (!has(state, IN_CLOSED)) {
+            state = unset(set(state, IN_CLOSED), READ);
+            final CloseEvent evt = CHANNEL_CLOSED_INBOUND;
+            storeCloseRequestAndEmit(evt);
+            inboundEventCheckClose(ctx.channel(), evt);
+        }
     }
 
     @Override
     void channelClosedOutbound(final ChannelHandlerContext ctx) {
-        state = unset(set(state, OUT_CLOSED), WRITE);
-        storeCloseRequestAndEmit(CHANNEL_CLOSED_OUTBOUND);
-        outboundEventCheckClose(ctx.channel());
+        if (!has(state, OUT_CLOSED)) {
+            state = unset(set(state, OUT_CLOSED), WRITE);
+            final CloseEvent evt = CHANNEL_CLOSED_OUTBOUND;
+            storeCloseRequestAndEmit(evt);
+            outboundEventCheckClose(ctx.channel(), evt);
+        }
+    }
+
+    @Override
+    void channelCloseNotify(final ChannelHandlerContext ctx) {
+        channelClosedInbound(ctx);
+        closeChannelOutbound(ctx.channel());
     }
 
     @Override
     void closeChannelInbound(final Channel channel) {
         state = set(state, IN_CLOSED);
         // todo storeCloseRequestAndEmit ?
-        inboundEventCheckClose(channel);
+        inboundEventCheckClose(channel, closeEvent);
     }
 
     @Override
     void closeChannelOutbound(final Channel channel) {
         state = set(state, OUT_CLOSED);
         // todo storeCloseRequestAndEmit ?
-        outboundEventCheckClose(channel);
+        outboundEventCheckClose(channel, closeEvent);
     }
 
     @Override
     void gracefulUserClosing(final Channel channel) {
         state = set(state, GRACEFUL_CLOSE);
-        storeCloseRequestAndEmit(GRACEFUL_USER_CLOSING);
+        final CloseEvent evt = GRACEFUL_USER_CLOSING;
+        storeCloseRequestAndEmit(evt);
         if (!isAnySet(state, READ_WRITE)) {
-            closeChannel(channel);
+            closeChannel(channel, evt);
         }
     }
 
-    private void inboundEventCheckClose(final Channel channel) {
+    private void inboundEventCheckClose(final Channel channel, @Nullable final CloseEvent evt) {
         if (isAllSet(state, OUT_CLOSED) || (isAnySet(state, GRACEFUL_IN_CLOSED) && !isAllSet(state, WRITE))) {
-            closeChannel(channel);
+            closeChannel(channel, evt);
         } else if (isAllSet(state, CLIENT_IN_WRITE)) {
             // If a client inbound has closed while writing we abort the write because we can't be sure if the write
             // will ever complete or receive any additional feedback form the server.
@@ -146,9 +165,9 @@ final class NonPipelinedCloseHandler extends CloseHandler {
         }
     }
 
-    private void outboundEventCheckClose(final Channel channel) {
+    private void outboundEventCheckClose(final Channel channel, @Nullable final CloseEvent evt) {
         if (isAllSet(state, IN_CLOSED) || (isAnySet(state, GRACEFUL_OUT_CLOSED) && !isAllSet(state, READ))) {
-            closeChannel(channel);
+            closeChannel(channel, evt);
         }
     }
 
@@ -159,9 +178,10 @@ final class NonPipelinedCloseHandler extends CloseHandler {
         eventHandler.accept(event);
     }
 
-    private void closeChannel(final Channel channel) {
+    private void closeChannel(final Channel channel, @Nullable final CloseEvent evt) {
         if (!has(state, CLOSED)) {
             state = set(state, ALL_CLOSED);
+            LOGGER.trace("{} Closing channel – evt: {}", channel, evt);
             channel.close();
         }
     }

--- a/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/RequestResponseCloseHandler.java
+++ b/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/RequestResponseCloseHandler.java
@@ -210,25 +210,45 @@ class RequestResponseCloseHandler extends CloseHandler {
     @Override
     void channelClosedInbound(final ChannelHandlerContext ctx) {
         assert ctx.executor().inEventLoop();
-        state = set(state, IN_CLOSED);
-        // Use the actual event that initiated graceful closure:
-        final CloseEvent evt = has(state, CLOSING_SERVER_GRACEFULLY) ? closeEvent : CHANNEL_CLOSED_INBOUND;
-        assert evt != null;
-        storeCloseRequestAndEmit(evt);
-        maybeCloseChannelOnHalfClosed(ctx.channel(), evt);
-        state = unset(state, READ);
+        if (!has(state, IN_CLOSED)) {
+            state = set(state, IN_CLOSED);
+            // Use the actual event that initiated graceful closure:
+            final CloseEvent evt = has(state, CLOSING_SERVER_GRACEFULLY) ? closeEvent : CHANNEL_CLOSED_INBOUND;
+            assert evt != null;
+            storeCloseRequestAndEmit(evt);
+            maybeCloseChannelOnHalfClosed(ctx.channel(), evt);
+            state = unset(state, READ);
+        }
     }
 
     @Override
     void channelClosedOutbound(final ChannelHandlerContext ctx) {
         assert ctx.executor().inEventLoop();
-        state = set(state, OUT_CLOSED);
-        storeCloseRequestAndEmit(CHANNEL_CLOSED_OUTBOUND);
-        if (!has(state, CLOSING_SERVER_GRACEFULLY)) {
-            // Only try to close when we are not closing server gracefully
-            maybeCloseChannelOnHalfClosed(ctx.channel(), CHANNEL_CLOSED_OUTBOUND);
+        if (!has(state, OUT_CLOSED)) {
+            state = set(state, OUT_CLOSED);
+            storeCloseRequestAndEmit(CHANNEL_CLOSED_OUTBOUND);
+            if (!has(state, CLOSING_SERVER_GRACEFULLY)) {
+                // Only try to close when we are not closing server gracefully
+                maybeCloseChannelOnHalfClosed(ctx.channel(), CHANNEL_CLOSED_OUTBOUND);
+            }
+            state = unset(state, WRITE);
         }
-        state = unset(state, WRITE);
+    }
+
+    @Override
+    void channelCloseNotify(final ChannelHandlerContext ctx) {
+        if (hasAny(state, OUT_CLOSED, CLOSING_SERVER_GRACEFULLY)) {
+            // We already closed outbound side of the channel, which triggers closure of SSLEngine and results in
+            // SslCloseCompletionEvent#SUCCESS event generated immediately. Connection is already in a closing state,
+            // we should ignore this event and wait for ChannelInputShutdownReadComplete from the remote peer.
+            return;
+        }
+
+        // Notify close handler first to enhance error reporting and prevent LB from selecting this connection.
+        channelClosedInbound(ctx);
+        // We MUST respond with a "close_notify" alert and close down the connection immediately, discarding any pending
+        // writes.
+        closeChannelOutbound(ctx.channel());
     }
 
     @Override

--- a/servicetalk-transport-netty-internal/src/test/java/io/servicetalk/transport/netty/internal/RequestResponseCloseHandlerTest.java
+++ b/servicetalk-transport-netty-internal/src/test/java/io/servicetalk/transport/netty/internal/RequestResponseCloseHandlerTest.java
@@ -93,6 +93,7 @@ import static io.servicetalk.transport.netty.internal.RequestResponseCloseHandle
 import static io.servicetalk.transport.netty.internal.RequestResponseCloseHandlerTest.Scenarios.Events.IC;
 import static io.servicetalk.transport.netty.internal.RequestResponseCloseHandlerTest.Scenarios.Events.ID;
 import static io.servicetalk.transport.netty.internal.RequestResponseCloseHandlerTest.Scenarios.Events.IE;
+import static io.servicetalk.transport.netty.internal.RequestResponseCloseHandlerTest.Scenarios.Events.IH;
 import static io.servicetalk.transport.netty.internal.RequestResponseCloseHandlerTest.Scenarios.Events.IS;
 import static io.servicetalk.transport.netty.internal.RequestResponseCloseHandlerTest.Scenarios.Events.OB;
 import static io.servicetalk.transport.netty.internal.RequestResponseCloseHandlerTest.Scenarios.Events.OC;
@@ -354,6 +355,7 @@ public class RequestResponseCloseHandlerTest {
                 outputShutdown.set(true);
                 LOGGER.debug("channel.shutdownOutput()");
                 h.channelClosedOutbound(ctx); // ChannelOutputShutdownEvent observed from transport
+                h.channelCloseNotify(ctx);  // SslHandler generates SslCloseCompletionEvent immediately
                 return future;
             });
             when(channel.close()).then(__ -> {
@@ -443,9 +445,15 @@ public class RequestResponseCloseHandlerTest {
                         break;
                     case IH:
                         order.verify(channel).shutdownInput();
+                        order.verify(h).channelClosedInbound(ctx);
                         break;
                     case OH:
                         order.verify(channel).shutdownOutput();
+                        order.verify(h).channelClosedOutbound(ctx);
+                        // Verify shutdownOutput() triggers channelCloseNotify, but actually it is no-op
+                        order.verify(h).channelCloseNotify(ctx);
+                        order.verify(h, never()).channelClosedInbound(ctx);
+                        order.verify(h, never()).closeChannelOutbound(channel);
                         break;
                     case FC:
                         order.verify(channel).close();
@@ -488,8 +496,14 @@ public class RequestResponseCloseHandlerTest {
                             verify(h, never()).protocolClosingOutbound(ctx);
                             break;
                         case IS:
+                            if (!events.contains(IH)) {
+                                verify(h, never()).channelClosedInbound(ctx);
+                            }
+                            break;
                         case OS:
-                            // These may be called implicitly, skip verify
+                            if (!events.contains(OH)) {
+                                verify(h, never()).channelClosedOutbound(ctx);
+                            }
                             break;
                         case SR:
                             verify(scc, never()).setSoLinger(0);

